src/main/java/nextstep/subway/domain/SubwayGraph.java

```diff
@@ -0,0 +1,63 @@
+package nextstep.subway.domain;
+
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.util.Assert;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SubwayGraph extends WeightedMultigraph<Station, SectionEdge> {
+    protected SubwayGraph(Class<? extends SectionEdge> edgeClass) {
+        super(edgeClass);
+    }
+
+    public SubwayPath findSubwayPath(Station sourceStation, Station targetStation) {
+        validateStation(sourceStation, targetStation);
+        DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath(this);
+        GraphPath path = dijkstraShortestPath.getPath(sourceStation, targetStation);
+        return getSubwayPath(path);
+    }
+
+    private void validateStation(Station sourceStation, Station targetStation) {
+        if (sourceStation == targetStation) {
```

Comment:
객체끼리 비교할 때는 ==을 쓰시면 안 됩니다.
구글에서 자바 동일성 동등성 이라는 키워드로 검색해보시기 바랍니다.

```suggestion
        if (sourceStation.equals(targetStation)) {
```

Reply:
이 부분은 JPA Entity 의 영속성 에 대한 동일성 비교 보장 때문에 == 를 사용하기는 했습니다!
equals 를 재정의 한 만큼 equals 로 통일하도록 하겠습니다!